src/main/java/org/xstefank/whitelist/WhitelistProcessing.java

```diff
@@ -0,0 +1,92 @@
+package org.xstefank.whitelist;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.xstefank.api.ContinuousIntegrationAPI;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.api.TeamCityAPI;
+import org.xstefank.model.PersistentList;
+import org.xstefank.model.Utils;
+import org.xstefank.model.yaml.FormatConfig;
+
+import static org.xstefank.model.Utils.getBooleanProperty;
+import static org.xstefank.model.Utils.WHITELIST_ENABLED;
+
+public class WhitelistProcessing {
+
+    public static final boolean IS_WHITELISTING_ENABLED = getBooleanProperty(WHITELIST_ENABLED);
+
+    private static final List<ContinuousIntegrationAPI> apis = loadApis();
+
+    private final PersistentList userList;
+    private final PersistentList adminList;
+
+    private final List<Command> commands;
+
+    public WhitelistProcessing(FormatConfig config) {
+        String dirName = System.getProperty(Utils.JBOSS_CONFIG_DIR);
+        userList = new PersistentList(dirName, Utils.USERLIST_FILE_NAME);
+        adminList = new PersistentList(dirName, Utils.ADMINLIST_FILE_NAME);
+        this.commands = getCommands(config);
+    }
+
+    public static boolean isUserEligibleForCIRun(JsonNode payload, PersistentList adminList) {
+        String username = payload.get(Utils.COMMENT).get(Utils.USER).get(Utils.LOGIN).asText();
+        return adminList.hasUsername(username);
+    }
+
+    public static String getAuthorUsernameFromIssuePayload(JsonNode payload) {
+        return payload.get(Utils.ISSUE).get(Utils.USER).get(Utils.LOGIN).asText();
+    }
+
+    public void process(JsonNode payload) {
+        if (payload.get(Utils.ISSUE).has(Utils.PULL_REQUEST) &&
+                payload.get(Utils.ACTION).asText().matches("created") &&
+                    !commands.isEmpty()) {
+            String message = payload.get(Utils.COMMENT).get(Utils.BODY).asText();
+            for (Command command : commands) {
+                if (message.matches(command.getCommandRegex())) {
+                    command.process(payload, adminList, userList);
+                }
+            }
+        }
+    }
+
+    private List<Command> getCommands(FormatConfig config) {
+        List<Command> commands = new ArrayList<>();
+
+        if (config.getFormat().getCommands() != null ||
```

Comment:
save `config.getFormat().getCommands()` prior to all these invocations, it is unnecessary to always call two methods to get to the same information